functions/Copy-DbaQueryStoreConfig.ps1

```diff
@@ -38,120 +54,116 @@ You should have received a copy of the GNU General Public License along with thi
+			
 		}
 		catch
 		{
-			if ($TargetSqlServer.count -eq 1)
-			{
-				throw $_
-			}
-			else
-			{
-				Write-Warning "Can't connect to $TargetSqlServer."
-				Continue
-			}
+				Write-Warning "Can't connect to $Destination."
+				continue
 		}
-
-        $sqlVersion = $targetserver.VersionMajor
-
-        if($sqlVersion -ilt "13")
-        {
-        Write-Warning "The SQL Server Instance ($TargetSqlServer) has a lower SQL Server version than SQL Server 2016. Skipping server."
-        continue
-        }
-
-        # We have to exclude all the system databases since they cannot have the Query Store feature enabled
-		$dbs = $targetserver.Databases | Where-Object { $_.name -ne 'TempDb' -and $_.name -ne 'master' -and $_.name -ne 'msdb' -and $_.name -ne 'model' }
-
-        if ($databases.count -gt 0)
+		
+		if ($sourceserver.VersionMajor -lt 13)
 		{
-			$dbs = $dbs | Where-Object { $databases -contains $_.Name }
+			Write-Warning "The source SQL Server Instance ($source) has a lower SQL Server version than SQL Server 2016. Skipping server."
+			continue
 		}
-
-		if ($exclude.count -gt 0)
+		
+		if ($destserver.VersionMajor -lt 13)
+		{
+			Write-Warning "The destination SQL Server Instance ($destination) has a lower SQL Server version than SQL Server 2016. Skipping server."
+			continue
+		}
+		
+		# We have to exclude all the system databases since they cannot have the Query Store feature enabled
+		$dbs = $destserver.Databases | Where-Object { $_.IsSystemObject -eq $false }
+		
+		
+		if ($DestinationDatabase.count -gt 0)
+		{
+			$dbs = $dbs | Where-Object { $DestinationDatabase -contains $_.Name }
+		}
+		
+		if ($Exclude.count -gt 0)
 		{
 			$dbs = $dbs | Where-Object { $exclude -notcontains $_.Name }
 		}
-
+		
+		if ($dbs.count -eq 0)
+		{
+			Write-Warning "No matching databases found. Check the spelling and try again."
+			return
+		}
+		
 		foreach ($db in $dbs)
 		{
-            $result = $null
-			Write-Verbose "Processing target database: $($db.name) on $targetservername"
-
+			Write-Verbose "Processing destination database: $db on $destination"
+	
 			if ($db.IsAccessible -eq $false)
 			{
-				Write-Warning "The database $($db.name) on server $targetservername is not accessible. Skipping database."
-				Continue
+				Write-Warning "The database $db on server $destination is not accessible. Skipping database."
+				continue
 			}
-            
-            # Set the Query Store configuration through the Set-DbaQueryStoreConfig function
-            Set-DbaQueryStoreConfig -SqlServer $TargetSqlServer -Databases $($db.name) -State $SourceQSConfig.ActualState -FlushInterval $SourceQSConfig.FlushInterval -CollectionInterval $SourceQSConfig.CollectionInterval -MaxSize $SourceQSConfig.MaxSize -CaptureMode $SourceQSConfig.CaptureMode -CleanupMode $SourceQSConfig.CleanupMode -StaleQueryThreshold $SourceQSConfig.StaleQueryThreshold
-                      
-		}  
-
-
-    }
-	
+			
```

Comment:
@ctrlbold I thought it might,
couldn't the version checks be left to Set- too then?